### PR TITLE
Added documentation for factorial of a number

### LINKED ISSUE #57: Documentation for factorial of a number
## 📚 Documentation

Documentation for factorial of a number.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes
Please assign this to me under GSSOC'21.

### PR DESCRIPTION
**Pull request by nidh-ish**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added documentation for factorial of a number.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #57

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -63,7 +63,8 @@ module.exports = {
       label: 'Mathematics',
       items: [
         'mathematics/mathematics',
-        'mathematics/armstrong-number'
+        'mathematics/armstrong-number',
+	'mathematics/factorial'
```
Comment (HarshCasper) [inline sidebars.js:67]: Can you indent it as per the format?

File: docs/mathematics/factorial.mdx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+```
+Time complexity - O(n)
+Space complexity - O(1)
+```
+
+## Credits
+
+   - [Harsh Ghodkar](https://github.com/Saviour1001) for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/cp/long_factorials.cpp)
+   - [Ricardo Prins](https://github.com/ricardoprins) for the [Python implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/math/factorial.py)
```
Comment (HarshCasper) [inline docs/mathematics/factorial.mdx:165]: This was worked on by Rutuja: https://github.com/TesseractCoding/NeoAlgo/pull/952

File: docs/mathematics/factorial.mdx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    factorial(num);
+    return 0;
+}
+```
+
+</TabItem>
+<TabItem value="py">
+
+```python
+def fact(x):
```
Comment (HarshCasper) [inline docs/mathematics/factorial.mdx:119]: Can you add a comment that we are using iteration here?

File: docs/mathematics/factorial.mdx [hunk 4]
```diff
@@ -0,0 +1,165 @@
+---
+title: Factorial
```
Comment (HarshCasper) [inline docs/mathematics/factorial.mdx:2]: It can be `Factorial of a Number`

File: docs/mathematics/factorial.mdx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+slug: /mathematics/Factorial
```
Comment (HarshCasper) [inline docs/mathematics/factorial.mdx:3]: Change the slug to be `factorial-of-number`

File: docs/mathematics/factorial.mdx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+---
+
+## Introduction
+
+Factorial of a number `n` is defined as the product of all positive integers less than or equal to `n`. Factorial function is defined for all non-negative integers. The value of `0!` is `1`.
+
+## Explaination
```
Comment (HarshCasper) [inline docs/mathematics/factorial.mdx:10]: It should be `Explanation`

File: docs/mathematics/factorial.mdx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+```
+Factorial(5) = 5 * 4 * 3 * 2 * 1 = 120
+Factorial(6) = 6 * 5 * 4 * 3 * 2 * 1 = 720 = 6 * Factorial(5)
+```
+
+## Algorithm
+
+### Recursive
+
+For a given number `n`,
```
Comment (HarshCasper) [inline docs/mathematics/factorial.mdx:35]: Can you give a stepwise explanation?

File: docs/mathematics/factorial.mdx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+factorial(n) = n * factorial(n - 1)
+```
+
+### Iterative
+
+For a given number `n`,
+
+Multiply numbers starting from `1` to `n` using a for loop.
+
+Return the factorial
```
Comment (HarshCasper) [inline docs/mathematics/factorial.mdx:47]: Make it stepwise

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/mathematics/factorial.mdx (hunk 2 above)
Comment (nidh-ish) [reply docs/mathematics/factorial.mdx:165]: No no that's different. This one which I used was implemented by Ricardo.

**Review by HarshCasper — APPROVED** [review]
LGTM!

File: docs/mathematics/factorial-of-number.mdx [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+}>
+
+<TabItem value="cpp">
+
+```cpp
+//The length of vector should be >= length of the resultant factorial
+#include<bits/stdc++.h>
+using namespace std;
+
+/*The function takes a number and muiltiplies the number in the school's old fashioned
```
Comment (vybhav72954) [inline docs/mathematics/factorial-of-number.mdx:84]: There are a few spelling mistakes here like `multiply` is spelled wrong. Kindly get them checked, also please cross heck your whole document once again.

**Review by ankitaggarwal23 — APPROVED** [review]

File: docs/mathematics/factorial-of-number.mdx [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+```
+
+### Iterative
+
+For a given number `n`,
+
+ - Declare a variable factorial and initialize it to `1`.
+ - For `num` starting from `1` to `n`.
+ - Compute `factorial` = `factorial` * `num`
+ - Add the end of the for loop return `factorial`.
```
Comment (vybhav72954) [inline docs/mathematics/factorial-of-number.mdx:55]: The statement sounds a bit vague to me.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
Kindly incorporate these changes, also please proofread the document once again. In case I will find something, will notify you here.

**Pull request merged**